### DISCUSSION TITLE
Where I can get the principle/algorithms of place and route？

### DISCUSSION THREAD

**Discussion by MbjYjbpivj** [tenstorrent/tt-mlir#395] [Q&A]
What are the principles and algorithms for partitioning subgraphs and routing? Where can I find reference materials?

**Comment by nsmithtt** [comment]
Hi @MbjYjbpivj, unlike Tenstorrent's TT-Buda compiler we are no longer placing operations spatially on the grid.  The tt-mlir project targets tt-metalium as a backend, this programming model documentation might be a good reference to see how operations coming out of the compiler actually get executed on hardware https://github.com/tenstorrent/tt-metal/blob/main/METALIUM_GUIDE.md
